src/core.rs

```diff
@@ -308,11 +341,11 @@ impl Fragment for Word<'_> {
         self.width as f64
     }
 
-    // We assume the whitespace consist of ' ' only. This allows us to
-    // compute the display width in constant time.
+    // Whitespace no longer consists of only spaces, so we have to
```

Comment:
This comment refers to the "past" and it makes sense when reading the diff in this PR. However, it won't make sense when reading the code in six months :-_